From 6e3fcd5d02cf5e0a9b1c4d42968e415bba503c6e Mon Sep 17 00:00:00 2001
From: Yanqin Wei <Yanqin.Wei@arm.com>
Date: Tue, 14 Feb 2023 14:40:20 +0800
Subject: [PATCH] soc: fvp_aemv8r: add mpu region from device tree

Some platforms need to define multiple memory regions with
various attribute. This patch adds dts defined regions in the mpu
configuration. The memory attribute can be set in the device tree.

Signed-off-by: Yanqin Wei <Yanqin.Wei@arm.com>
Upstream-Status: Backport [main:a1f55c63b37dd642733510364186184a3bfa5861]
---
 soc/arm64/arm/fvp_aemv8r/arm_mpu_regions.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/soc/arm64/arm/fvp_aemv8r/arm_mpu_regions.c b/soc/arm64/arm/fvp_aemv8r/arm_mpu_regions.c
index 22b4822360b3..97b08d418b4f 100644
--- a/soc/arm64/arm/fvp_aemv8r/arm_mpu_regions.c
+++ b/soc/arm64/arm/fvp_aemv8r/arm_mpu_regions.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Arm Limited (or its affiliates). All rights reserved.
+ * Copyright (c) 2021-2023 Arm Limited (or its affiliates). All rights reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
 
@@ -7,6 +7,7 @@
 
 #include <zephyr/arch/arm64/cortex_r/arm_mpu.h>
 #include <zephyr/linker/linker-defs.h>
+#include <zephyr/linker/devicetree_regions.h>
 #include <zephyr/sys/util.h>
 
 #define DEVICE_REGION_START 0x80000000UL
@@ -46,7 +47,10 @@ static const struct arm_mpu_region mpu_regions[] = {
 	MPU_REGION_ENTRY("DEVICE",
 			 DEVICE_REGION_START,
 			 DEVICE_REGION_END,
-			 REGION_DEVICE_ATTR)
+			 REGION_DEVICE_ATTR),
+
+	/* Extra regions defined in device tree */
+	LINKER_DT_REGION_MPU(MPU_REGION_ENTRY_FROM_DTS)
 };
 
 const struct arm_mpu_config mpu_config = {
-- 
2.34.1

